api/payments.go

```diff
@@ -276,18 +284,17 @@ func (a *API) PaymentRefund(ctx context.Context, w http.ResponseWriter, r *http.
 		return
 	}
 
-	provider := gcontext.GetPaymentProvider(ctx)
+	provider := gcontext.GetPaymentProviders(ctx)[order.PaymentProcessor]
```

Comment:
I don't see a check if order.PaymentProcessor is empty. It might also be worth normalizing the param